### PR TITLE
remove kubeadmin workaround

### PR DESCRIPTION
**Pull request by SherinV**
Signed-off-by: Sherin Varughese <shvarugh@redhat.com>

**Related Issue:**  [stolostron/backlog#25996](https://github.com/stolostron/backlog/issues/25996)

### Description of changes
Remove rbac workaround for kubeadmin
For users to get the correct SSRR, all provided user info should be passed to impersonation config.
Just passing username and UID will not suffice to create the correct SSRR, if other info like groups and extra are present in token review.

### CODE REVIEW

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

File: pkg/rbac/userData.go [hunk 1]
```diff
@@ -258,6 +258,9 @@ func (user *UserDataCache) getNamespacedResources(cache *Cache, ctx context.Cont
 		} else {
 			klog.V(9).Infof("SelfSubjectRulesReviews Kube API result for ns:%s : %v\n", ns, prettyPrint(result.Status))
 		}
+
+		// Name the loop - to break out of the loop if user is kubeadmin (have access to everything)
```
Comment (jlpadilla) [inline pkg/rbac/userData.go:262]: The user doesn't have to be 'kubeadmin', other users could have similar `*` access.
```suggestion
		// Name the loop - to break out of the loop if user has access to everything (*)
```

Comment (SherinV) [reply pkg/rbac/userData.go]: Updated

File: pkg/rbac/userData_test.go [hunk 2]
```diff
@@ -610,3 +611,81 @@ func Test_getUserData(t *testing.T) {
...
+		userData:     UserData{},
+		nsrUpdatedAt: time.Now(),
+	}
+	_, err := udc.getImpersonationClientSet("123456", mock_cache)
+	// Ensure that there is no error
+	assert.Nil(t, err)
+
+}
+
+func Test_getNamespaces_kubeAdmin(t *testing.T) {
```
Comment (jlpadilla) [inline pkg/rbac/userData_test.go:646]: Rename test.
```suggestion
// User has access to all resources in the namespace.
func Test_hasAccessToAllResourcesInNamespace(t *testing.T) {
```

**Review by jlpadilla — APPROVED** [review]
/lgtm

**Pull request merged**